Upgrade all deployment platforms to Ubuntu 14.04.5

We are using Ubuntu 14.04.2 on Vagrant and AWS at the moment. [Travis CI is in the process of moving to the latest version of 14.04.5](https://blog.travis-ci.com/2017-06-19-trusty-updates-2017-Q2) ~~and it is causing builds to fail~~.

Ubuntu 14.04.5 is the latest in the 14.04 LTS series, while 14.04.2 is almost 2.5 years old at this point (https://wiki.ubuntu.com/Releases).

We should standardize on Ubuntu 14.04.5 across all deployment platforms (Vagrant, Travis and AWS).

We should probably [constrain](https://www.vagrantup.com/docs/boxes/versioning.html#version-constraints) Vagrant environment to a specific version of the Ubuntu box using the Vagrantfile by specifying the `config.vm.box_version` option.

FWIW, my [build](https://travis-ci.org/refinery-platform/refinery-platform/builds/245479188) just failed:
```
$ bower install --config.interactive=false --quiet
module.js:472
    throw err;
    ^

Error: Cannot find module 'internal/fs'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at evalmachine.<anonymous>:18:20
    at Object.<anonymous> (/home/travis/.nvm/versions/node/v7.4.0/lib/node_modules/bower/lib/node_modules/graceful-fs/fs.js:11:1)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)


The command "bower install --config.interactive=false --quiet" failed and exited with 1 during .

Your build has been stopped.
```

--- a/deployment/terraform/modules/ec2/main.tf
+++ b/deployment/terraform/modules/ec2/main.tf
@@ -119,7 +119,7 @@ resource "aws_iam_instance_profile" "app_server" {
 
 resource "aws_instance" "app_server" {
   count                  = "${var.instance_count}"
-  ami                    = "ami-d05e75b8"
+  ami                    = "ami-03597b1b84c02cf7b"
   instance_type          = "${var.instance_type}"
   key_name               = "${var.key_pair_name}"
   monitoring             = true

--- a/deployment/terraform/modules/ec2/main.tf
+++ b/deployment/terraform/modules/ec2/main.tf
@@ -119,6 +119,7 @@ resource "aws_iam_instance_profile" "app_server" {
 
 resource "aws_instance" "app_server" {
   count                  = "${var.instance_count}"
+  # ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-20181203
   ami                    = "ami-03597b1b84c02cf7b"
   instance_type          = "${var.instance_type}"
   key_name               = "${var.key_pair_name}"

